src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -166,7 +166,9 @@ public void DeleteSearchParameter(ITypedElement searchParam)
 
             if (!UrlLookup.TryRemove(new Uri(searchParamWrapper.Url), out searchParameterInfo))
             {
+#pragma warning disable CA2201
```

Comment:
Can we add a comment for why we are disabling the rule here?

Reply:
@feordin
is there are any better solution we can throw here?

Reply:
Perhaps we can throw a ResourceNotFoundException.